src/components/ToggleSwitch/index.tsx

```diff
@@ -0,0 +1,210 @@
+import * as React from 'react';
+import {
+  StyleSheet,
+  Animated,
+  TouchableOpacity,
+  Easing,
+  StyleProp,
+  Text,
+  TextStyle,
+  Platform,
+  TouchableOpacityProps
+} from 'react-native';
+import dimen from '../../utils/dimen';
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    alignItems: 'center'
+  },
+  basicStyle: {
+    height: dimen.normalizeDimen(16),
+    width: dimen.normalizeDimen(16),
+    borderRadius: dimen.normalizeDimen(16 / 2),
+    backgroundColor: '#2C67BC'
+  },
+  eahcStyles: {
+    fontSize: dimen.normalizeDimen(12),
+    color: '#f4f3f4',
+    position: 'absolute',
+    top: dimen.normalizeDimen(2),
+    left: dimen.normalizeDimen(3)
+  },
+
+  eachStylesOff: {
+    fontSize: dimen.normalizeDimen(12),
+    color: '#f4f3f4',
+    position: 'absolute',
+    top: dimen.normalizeDimen(2),
+    right: dimen.normalizeDimen(3)
+  },
+  mainStyes: {
+    borderRadius: dimen.normalizeDimen(12),
+    backgroundColor: '#F5F5F5',
+    height: dimen.normalizeDimen(20),
+    width: dimen.normalizeDimen(42),
+    padding: dimen.normalizeDimen(2),
+    ...Platform.select({
+      ios: {
+        shadowColor: 'rgba(0,0,0,0.25)',
+        shadowOffset: {width: 0, height: 0},
+        shadowOpacity: dimen.normalizeDimen(1),
+        shadowRadius: dimen.normalizeDimen(1)
+      },
+      android: {
+        elevation: dimen.normalizeDimen(1)
+      }
+    })
+  },
+  labelLeft: {
+    marginRight: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  },
+  labelRight: {
+    marginLeft: 5,
+    color: '#FFFFFF',
+    fontSize: 12,
+    fontWeight: '400'
+  }
+});
+
+interface ToggleSwitchProps {
+  value?: boolean;
+  labelLeft?: string;
+  labelRight?: string;
+  circleActiveColor?: string;
+  circleInActiveColor?: string;
+  backgroundActive?: string;
+  backgroundInactive?: string;
+  styleLabelLeft?: StyleProp<TextStyle>;
+  styleLabelRight?: StyleProp<TextStyle>;
+  activeTextColor?: string;
+  inactiveTextColor?: string;
+  containerStyle?: TouchableOpacityProps['style'];
+  labelOff?: string;
+  labelOn?: string;
+  onValueChange?: (value: boolean) => void;
+}
+
+const ToggleSwitch: React.FC<ToggleSwitchProps> = ({
+  value = false,
+  onValueChange,
+  labelLeft,
+  labelRight,
+  circleActiveColor = '#00ADB5',
+  circleInActiveColor = '#2C67BC',
+  backgroundActive = '#F5F5F5',
+  backgroundInactive = '#F5F5F5',
+  styleLabelLeft,
+  styleLabelRight,
+  activeTextColor = '#00ADB5',
+  inactiveTextColor = '#2C67BC',
+  containerStyle,
+  labelOff = 'Off',
+  labelOn = 'On'
+}) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The default values for the props are hardcoded in the component. It would be better to define these as constants outside of the component, which would make it easier to manage and change them if needed.
```diff
+ const DEFAULT_VALUES = {
+   value: false,
+   circleActiveColor: '#00ADB5',
+   circleInActiveColor: '#2C67BC',
+   backgroundActive: '#F5F5F5',
+   backgroundInactive: '#F5F5F5',
+   activeTextColor: '#00ADB5',
+   inactiveTextColor: '#2C67BC',
+   labelOff: 'Off',
+   labelOn: 'On'
+ };

const ToggleSwitch: React.FC<ToggleSwitchProps> = ({
-   value = false,
-   onValueChange,
-   labelLeft,
-   labelRight,
-   circleActiveColor = '#00ADB5',
-   circleInActiveColor = '#2C67BC',
-   backgroundActive = '#F5F5F5',
-   backgroundInactive = '#F5F5F5',
-   styleLabelLeft,
-   styleLabelRight,
-   activeTextColor = '#00ADB5',
-   inactiveTextColor = '#2C67BC',
-   containerStyle,
-   labelOff = 'Off',
-   labelOn = 'On'
+   value = DEFAULT_VALUES.value,
+   onValueChange,
+   labelLeft,
+   labelRight,
+   circleActiveColor = DEFAULT_VALUES.circleActiveColor,
+   circleInActiveColor = DEFAULT_VALUES.circleInActiveColor,
+   backgroundActive = DEFAULT_VALUES.backgroundActive,
+   backgroundInactive = DEFAULT_VALUES.backgroundInactive,
+   styleLabelLeft,
+   styleLabelRight,
+   activeTextColor = DEFAULT_VALUES.activeTextColor,
+   inactiveTextColor = DEFAULT_VALUES.inactiveTextColor,
+   containerStyle,
+   labelOff = DEFAULT_VALUES.labelOff,
+   labelOn = DEFAULT_VALUES.labelOn
}) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->